Merge established changes from Unstable into Alpha

pacharanero opened the pull request.

eatyourpeas committed:
references broken up

eatyourpeas committed:
add decimal ages to uk-who-preterm

eatyourpeas committed:
refactored references into reference-specific files, function to select in sds_calculations.py

eatyourpeas committed:
new uk_who_sds_calcuation function

eatyourpeas committed:
refactor decimal ages added to references

eatyourpeas committed:
functioning uk_who_sds_calculation

eatyourpeas committed:
age indices removed

eatyourpeas committed:
percentage median bmi works

eatyourpeas committed:
refactor sds_calculation to uk_who.py. create global functions.

eatyourpeas committed:
includes trisomy 21

eatyourpeas committed:
add turner

eatyourpeas committed:
pytest passed

eatyourpeas committed:
update comments

eatyourpeas committed:
bug fix

eatyourpeas committed:
fix for flask chart data generation

pacharanero committed:
Merge pull request #106 from rcpch/referencesrefactor

Referencesrefactor

eatyourpeas committed:
Refactor Measurement Class to accept multiple references
Fixes #107

eatyourpeas committed:
Merge pull request #108 from rcpch/eatyourpeas/issue107

Refactor Measurement Class to accept multiple references

eatyourpeas committed:
Refactor Measurement Value
Fixes #109

eatyourpeas committed:
Merge pull request #111 from rcpch/eatyourpeas/issue109

Refactor Measurement Value

eatyourpeas committed:
intial refactor

eatyourpeas committed:
measurement_from_sds needs to be agnostic of the reference it is applied to
Fixes #112

pacharanero committed:
Merge pull request #113 from rcpch/measurement_for_sds

Measurement for sds.
Tested on Postman and Pytest

eatyourpeas committed:
Percentage median BMI now fixed
Includes error handling for all references
Fixes #114

eatyourpeas committed:
Merge pull request #115 from rcpch/eatyourpeas/issue114

Percentage median BMI now fixed

eatyourpeas committed:
create new controllers

eatyourpeas committed:
create_chart function agnostic of reference implemented.
another step towards We should be able to produce growth charts for Down's Syndrome #41

eatyourpeas committed:
Merge pull request #116 from rcpch/eatyourpeas/issue41

Eatyourpeas/issue41

pacharanero committed:
experimental addition of pytest step in the GitHub Action

pacharanero committed:
Merge commit 'a2360dd187f6294487fcbd09ff5adba47bb023d8' into unstable

pacharanero committed:
experimental take 2 testing in the Action

pacharanero committed:
deliberately broken Measurement class so that the tests will fail on GitHub Actions (testing the tests)

pacharanero committed:
Reverted the deliberately broken Measurement class, now that test dploy of broken class proved that the GH deploy process fails if the tests fail.

pacharanero committed:
Adds on-deploy tests to the alpha branch too

pacharanero committed:
Changed reference names to lowercase. Miscellaneous linter formatting improvements.

eatyourpeas committed:
validation included

eatyourpeas committed:
validation refactor with blueprints

pacharanero committed:
moves Plottable Child into the UK-WHO blueprint

pacharanero committed:
disable some tests until I have time to implement them properly

pacharanero committed:
some tidying

pacharanero committed:
removes 'SingleCalculation*' in favour of just 'Calculation*'

eatyourpeas committed:
functions chart data generation

eatyourpeas committed:
create_chart fix

eatyourpeas committed:
removed save chart data

pacharanero committed:
Refactored the openAPI3 JSON and YAML generator out into a Blueprint. app.py is a calm and tidy place now

pacharanero committed:
Merge branch 'unstable' of github.com:rcpch/digital-growth-charts-server into unstable

eatyourpeas committed:
fix chart data return object

eatyourpeas committed:
merge with open api

pacharanero committed:
This is a fix for yesterday's refactor which accidentally messed up APIspec Autogeneration.

pacharanero committed:
Merge branch 'unstable' of github.com:rcpch/digital-growth-charts-server into unstable

pacharanero committed:
some further tweaks to the Docker setup

pacharanero committed:
readme additions for Docker in development

pacharanero committed:
updates to build docker scfipt

eatyourpeas committed:
create all charts and tims centiles

eatyourpeas committed:
create charts

eatyourpeas committed:
redo charts

eatyourpeas committed:
create uk-who charts at runtime

eatyourpeas committed:
autogenerated charts for all references on load

eatyourpeas committed:
label to age_type in plottablechild

eatyourpeas committed:
plottable child changes

eatyourpeas committed:
update

eatyourpeas committed:
comment out growth chart data generation on deploy

eatyourpeas committed:
update measurement class return corrected calendar age

eatyourpeas committed:
add measurement_method to plottable child return object

eatyourpeas committed:
update readme

eatyourpeas committed:
tidy corrected age

statist7 commented:
Wow! - someone‘s been busy!

Sent from my iPhone

On 6 Feb 2021, at 12:58, Marcus Baw <notifications@github.com> wrote:

﻿
________________________________

You can view, comment on, or merge this pull request online at:

  https://github.com/rcpch/digital-growth-charts-server/pull/129

Commit Summary

  *   references broken up
  *   add decimal ages to uk-who-preterm
  *   refactored references into reference-specific files, function to select in sds_calculations.py
  *   new uk_who_sds_calcuation function
  *   refactor decimal ages added to references
  *   functioning uk_who_sds_calculation
  *   age indices removed
  *   percentage median bmi works
  *   refactor sds_calculation to uk_who.py. create global functions.
  *   includes trisomy 21
  *   add turner
  *   pytest passed
  *   update comments
  *   bug fix
  *   fix for flask chart data generation
  *   Merge pull request #106 from rcpch/referencesrefactor
  *   Refactor Measurement Class to accept multiple references
  *   Merge pull request #108 from rcpch/eatyourpeas/issue107
  *   Refactor Measurement Value
  *   Merge pull request #111 from rcpch/eatyourpeas/issue109
  *   intial refactor
  *   measurement_from_sds needs to be agnostic of the reference it is applied to
  *   Merge pull request #113 from rcpch/measurement_for_sds
  *   Percentage median BMI now fixed
  *   Merge pull request #115 from rcpch/eatyourpeas/issue114
  *   create new controllers
  *   create_chart function agnostic of reference implemented.
  *   Merge pull request #116 from rcpch/eatyourpeas/issue41
  *   experimental addition of pytest step in the GitHub Action
  *   Merge commit 'a2360dd187f6294487fcbd09ff5adba47bb023d8' into unstable
  *   experimental take 2 testing in the Action
  *   deliberately broken Measurement class so that the tests will fail on GitHub Actions (testing the tests)
  *   Reverted the deliberately broken Measurement class, now that test dploy of broken class proved that the GH deploy process fails if the tests fail.
  *   Adds on-deploy tests to the alpha branch too
  *   Changed reference names to lowercase. Miscellaneous linter formatting improvements.
  *   validation included
  *   validation refactor with blueprints
  *   moves Plottable Child into the UK-WHO blueprint
  *   disable some tests until I have time to implement them properly
  *   some tidying
  *   removes 'SingleCalculation*' in favour of just 'Calculation*'
  *   functions chart data generation
  *   create_chart fix
  *   removed save chart data
  *   Refactored the openAPI3 JSON and YAML generator out into a Blueprint. app.py is a calm and tidy place now
  *   Merge branch 'unstable' of github.com:rcpch/digital-growth-charts-server into unstable
  *   fix chart data return object
  *   merge with open api
  *   This is a fix for yesterday's refactor which accidentally messed up APIspec Autogeneration.
  *   Merge branch 'unstable' of github.com:rcpch/digital-growth-charts-server into unstable
  *   some further tweaks to the Docker setup
  *   readme additions for Docker in development
  *   updates to build docker scfipt
  *   create all charts and tims centiles
  *   create charts
  *   redo charts
  *   create uk-who charts at runtime
  *   autogenerated charts for all references on load
  *   label to age_type in plottablechild
  *   plottable child changes
  *   update
  *   comment out growth chart data generation on deploy
  *   update measurement class return corrected calendar age
  *   add measurement_method to plottable child return object
  *   update readme
  *   tidy corrected age

File Changes

  *   M .github/workflows/alpha_rcpch-dgc-server-alpha.yml<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-4c22d0de80c5087b669348076675fab6dbe0fa1a0ae0bd50cc18ea9f550e2cc8> (47)
  *   M .github/workflows/unstable_rcpch-dgc-server-unstable.yml<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-4af47dd1cbee67644da9fbf6798e1f4b9ed6042787836ba4af57f2568fd9c12e> (47)
  *   M .gitignore<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947> (5)
  *   M Dockerfile<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557> (11)
  *   M README.md<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5> (26)
  *   A apispec_generation.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-382dcb73280c810dc9b42813fa8017ba933af740579114239926dd7b08443144> (94)
  *   M app.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-568470d013cd12e4f388206520da39ab9a4e4c3c6b95846cbc281abc1ba3c959> (333)
  *   M blueprints/__init__.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-6a871e951adf07b320d1c0babe4801a61437a6c3b7a0c9f6785293a616f104dd> (5)
  *   A blueprints/openapi_blueprint.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-c0e40745bf2822d9ab689e60cbbbe89b882f0d273c15e24c3c81e430c71d8bee> (38)
  *   A blueprints/uk_who_blueprint.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-ea1533b2dd06facc0987b6542876ecb95b69601019133f62806ee3c71df57f7b> (175)
  *   M controllers/__init__.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-99acfc9e804ef000aafc8020f5f2c29802fc8b5c998fc4f2fc318a6fe2a0c1bf> (1)
  *   M controllers/calculations.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-eeaf6080a25d7f14fc5932468d30ddcc6f30d847a271fcb58154c1af6c8b0ef9> (22)
  *   D controllers/chart.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-d9e867723cf2f1ef5676c4430e107dba8b13d35887baaeefb0e389b2c4a207ba> (279)
  *   M controllers/fictional_children.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-a38dded4b60b2b38f7fc82ebf116efa70a268fd0bb593dd47023e7778c274b8f> (6)
  *   M controllers/plottable_child.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-94cd9cc55bc37ed7b77d8bb4f8916775cc2b3b6838b378a163fe7d73a7b427c6> (70)
  *   M controllers/temp_test_functions.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-352ba0552a4bc711d7b9ec4bdc344e658edccc37073d77e1ddcd0b36b082168b> (8)
  *   A controllers/uk_who_chart.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-9b499cecaf691f607347874de82b3c9f3ffdee70c834ebe8503189b5ab697de7> (32)
  *   M controllers/upload.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-e14eeb08847460a7f55e934a3e21a8c17dd4e12b35bc3b55583620e4f17064b7> (240)
  *   M docs/developer-documentation/README.md<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-0a843bb476fefe424de57bacbc0e6cfb5fd457b252ff3611eaaf8eb890953c55> (17)
  *   A openapi.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-a9865368a7fc7fa33065e35b2343f10d08fb79d65205435403d0a163a3044713> (523)
  *   R openapi.yml<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-5016bdc926b6c0f649f363739d09159a1db664aae19a08fa6b38b03406f040dc> (144)
  *   M rcpchgrowth.code-workspace<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-41156f2e03b9f5ca0a1837449a790def4bae87eb38df30ed6160b9adf528b5d3> (7)
  *   M rcpchgrowth/rcpchgrowth/__init__.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-f6bdded4fed72f92255b5bcf5c968a5f060660f17394527f2ec84e02bf25f164> (4)
  *   M rcpchgrowth/rcpchgrowth/bmi_functions.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-5a260ef63e431bc535c5b4300358760c94d3d28c17464a470de608118dd3182a> (5)
  *   M rcpchgrowth/rcpchgrowth/constants/__init__.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-e22bc786ac37e92a16dcfb83e85e3faa9152cd2e5f207aa95c508cf97a7be324> (1)
  *   M rcpchgrowth/rcpchgrowth/constants/measurement_constants.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-9ed9951787b960d443706fc97526d94af0a5f7d035f78270217ed1fe2900a7b5> (74)
  *   A rcpchgrowth/rcpchgrowth/constants/parameter_constants.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-e531335b8ed85f4c17ad11ccb9e405daf3116cd7debd571d638133787ab907bd> (22)
  *   M rcpchgrowth/rcpchgrowth/constants/validation_constants.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-134b5894cf5b11d94c1e0045168f73827eaab205b604bcb3ccbba7bdb6ef18f4> (6)
  *   M rcpchgrowth/rcpchgrowth/data_tables/data_tables.txt<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-0df77b164b86164412392f033654be93ae65c0fe3075414a787ae9605af853bd> (1)
  *   M rcpchgrowth/rcpchgrowth/data_tables/trisomy_21.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-4596a2bef7c47cea4f40811f1d6f0ecfabef04ecb47d0426c37179c4cb0eb61a> (3803)
  *   M rcpchgrowth/rcpchgrowth/data_tables/turner.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-e2e31993006eec798b3f187baa0d748a729ca499a0d8e51c1e7069e36a949d33> (43)
  *   R rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/GrowthCharts.xls<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-84f4024f917b9dcbf3ac42b4df5afc0e6c0c25406097320b23410cffbed9068e> (0)
  *   R rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/RCPCH weight correlation matrix by month.csv<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-212070e5c06f5b055df91065b02d945d4364ff8c5ee1c75ee33060df88756950> (0)
  *   R rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/RCPCH weight correlation matrix by week.csv<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-9a5c14b3fffd9e35934911a50eee1d225fbdbe2c0d705c95bd8c0afe2cc03c4b> (0)
  *   A rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/huiqi_cole_methods.bas<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-f07adfd45e7ade409988769d4ec0221830df953e87718c1737e2f9049a6b4b5b> (4933)
  *   R rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/uk_who_0_20_preterm.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-fc6185a2d0a9e92380e7dffd3ae9b641a437d0732969dd8a1ca549f5291b31c5> (0)
  *   R rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/uk_who_0_20_term.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-b60839751e0d9f292b81f3dd360a0d4d793463d73ef58bebffbfc87ffd64b07d> (0)
  *   A rcpchgrowth/rcpchgrowth/data_tables/uk90_child.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-9c749f32d48bfea45ad9a7ae9fb358e60742a3a74227bc090f9f845dbb146672> (1517)
  *   A rcpchgrowth/rcpchgrowth/data_tables/uk90_preterm.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-078a6a36e2981c9d37d5a6f25614cd09e1f60ad4e755de209466d9addd8ffc86> (189)
  *   A rcpchgrowth/rcpchgrowth/data_tables/uk90_term.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-892ff9c5483cbed7e4fb22ca604fdafeaa8eed7bdbf9c737d48625aacc2863ae> (41)
  *   D rcpchgrowth/rcpchgrowth/data_tables/uk_who.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-94ead9ef676dab12026cd6a9c24c5a440115359354ba139a318c636c1a84d5e4> (32)
  *   D rcpchgrowth/rcpchgrowth/data_tables/uk_who_0_20_preterm_intervals_included.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-e1f0d1f2c54f48d60fc13df990aec541f75a21b6c158d808c06283593d1df14f> (2160)
  *   D rcpchgrowth/rcpchgrowth/data_tables/uk_who_centiles.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-28e84629c89e6de5ddbde09bc1dbb3c745f83ef26aad2ca5ad292d2f55be160e> (0)
  *   A rcpchgrowth/rcpchgrowth/data_tables/who_children.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-2723a52dd474ee88d932c508e4ec58f32b3245b0f78ba93febd5582a1550eadb> (233)
  *   A rcpchgrowth/rcpchgrowth/data_tables/who_infants.json<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-5578d9a4d81a830bf38a4b48f7a61d26bea22d1820c10d58c585584391c5658a> (323)
  *   M rcpchgrowth/rcpchgrowth/date_calculations.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-a4bdbd58a757a0fad02eaa661b491d7944859f1a999e8c744c5e0e434e219341> (28)
  *   M rcpchgrowth/rcpchgrowth/dynamic_growth.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-a43be1b2fea5f8df4f3afe380f91a2ed3fad7b9e868cc71482ebc81911987fd2> (179)
  *   M rcpchgrowth/rcpchgrowth/fictional_children.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-027d24fbe1ac44a24a7d62127d2ba684408642b0c20af166c61f33bb4466ebac> (5)
  *   A rcpchgrowth/rcpchgrowth/global_functions.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-319f1da7c00841a162ee41ed3da11847d063cbaf7b912d386536ccddca057b8f> (648)
  *   M rcpchgrowth/rcpchgrowth/measurement.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-1070ce6b9a9a5be3c3a02a7a664e1165657555392ebe15958ef8dda0ceb7fd29> (299)
  *   A rcpchgrowth/rcpchgrowth/schemas/__init__.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-d971d62c38a7cbde6e9b11fb59e2e3738f4bdb98fec7909cdf9faf24d1341113> (1)
  *   A rcpchgrowth/rcpchgrowth/schemas/measurement_class_schema.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-ec62ed2a261f99a7a13ab82549189ae9db98279ced92fbf7b42569ef5b136c47> (29)
  *   D rcpchgrowth/rcpchgrowth/sds_calculations.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-320b5b8a7c7499189dece810c52f8af00e598be3eb978ddf7299713c4e0cb515> (483)
  *   M rcpchgrowth/rcpchgrowth/tests/test_measurement_class.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-a09bbf79fc7515be0365e7892936b39db734c25e47df5ab6d042c759d645a986> (32)
  *   A rcpchgrowth/rcpchgrowth/trisomy_21.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-8b78a41727c2adaa37b79639ff3177148d5dcbdcf9fe05d7092fe65000493dcc> (75)
  *   A rcpchgrowth/rcpchgrowth/turner.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-f654fb5d6ea53476a05d1088eeeb556c8feb7e6223aca922b967947b1f3ee15a> (77)
  *   A rcpchgrowth/rcpchgrowth/uk_who.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-d432f60104a0f126a8158e08c21fcff35d3f1c0a9444a1d2385fe28608c56b11> (216)
  *   M rcpchgrowth/setup.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-245bf863fb42663de9544caa6bd82cb585ff0cf03074f0bf8c78454a4fc3375d> (7)
  *   M requirements.txt<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552> (6)
  *   A s/rebuild-docker<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-d0bcee849b6beac8ad46d6866e54ccff6b87f091cafdbebcaa84f00717a00193> (20)
  *   M s/start-docker<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-5e3fbf896a54e69f44a77ddcff819be82ef17c0be7ff4292b98035e2b4123ff8> (14)
  *   M schemas/__init__.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-1d41be12fbded48f0ac94e7231190b1752403326c3d0bf98792522a1162be0f0> (2)
  *   R schemas/calculation_schemas.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-42091a942d3be09918b5880dae4122d069fbe9656ae258a080b30d72130d51a5> (13)
  *   M schemas/measurement_schemas.py<https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-ea2ef5c4f0cdcb5ecbe8dc31960e522324084b5abebf10cc891311f1d83af462> (2)

Patch Links:

  *   https://github.com/rcpch/digital-growth-charts-server/pull/129.patch
  *   https://github.com/rcpch/digital-growth-charts-server/pull/129.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/rcpch/digital-growth-charts-server/pull/129>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGSJJIM54SNTSTXWQWSZADS5U4IDANCNFSM4XGII6EA>.

pacharanero commented:
It's about 3 months of development on the server, that was on Unstable but
not yet pulled into Alpha!
That was the cause of the weird issue on the growth.rcpch.ac.uk site. All
fixed now!

On Sat, 6 Feb 2021 at 13:22, Tim Cole <notifications@github.com> wrote:

> Wow! - someone‘s been busy!
>
> Sent from my iPhone
>
> On 6 Feb 2021, at 12:58, Marcus Baw <notifications@github.com> wrote:
>
> ﻿
> ________________________________
>
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/rcpch/digital-growth-charts-server/pull/129
>
> Commit Summary
>
> * references broken up
> * add decimal ages to uk-who-preterm
> * refactored references into reference-specific files, function to select
> in sds_calculations.py
> * new uk_who_sds_calcuation function
> * refactor decimal ages added to references
> * functioning uk_who_sds_calculation
> * age indices removed
> * percentage median bmi works
> * refactor sds_calculation to uk_who.py. create global functions.
> * includes trisomy 21
> * add turner
> * pytest passed
> * update comments
> * bug fix
> * fix for flask chart data generation
> * Merge pull request #106 from rcpch/referencesrefactor
> * Refactor Measurement Class to accept multiple references
> * Merge pull request #108 from rcpch/eatyourpeas/issue107
> * Refactor Measurement Value
> * Merge pull request #111 from rcpch/eatyourpeas/issue109
> * intial refactor
> * measurement_from_sds needs to be agnostic of the reference it is applied
> to
> * Merge pull request #113 from rcpch/measurement_for_sds
> * Percentage median BMI now fixed
> * Merge pull request #115 from rcpch/eatyourpeas/issue114
> * create new controllers
> * create_chart function agnostic of reference implemented.
> * Merge pull request #116 from rcpch/eatyourpeas/issue41
> * experimental addition of pytest step in the GitHub Action
> * Merge commit 'a2360dd187f6294487fcbd09ff5adba47bb023d8' into unstable
> * experimental take 2 testing in the Action
> * deliberately broken Measurement class so that the tests will fail on
> GitHub Actions (testing the tests)
> * Reverted the deliberately broken Measurement class, now that test dploy
> of broken class proved that the GH deploy process fails if the tests fail.
> * Adds on-deploy tests to the alpha branch too
> * Changed reference names to lowercase. Miscellaneous linter formatting
> improvements.
> * validation included
> * validation refactor with blueprints
> * moves Plottable Child into the UK-WHO blueprint
> * disable some tests until I have time to implement them properly
> * some tidying
> * removes 'SingleCalculation*' in favour of just 'Calculation*'
> * functions chart data generation
> * create_chart fix
> * removed save chart data
> * Refactored the openAPI3 JSON and YAML generator out into a Blueprint.
> app.py is a calm and tidy place now
> * Merge branch 'unstable' of github.com:rcpch/digital-growth-charts-server
> into unstable
> * fix chart data return object
> * merge with open api
> * This is a fix for yesterday's refactor which accidentally messed up
> APIspec Autogeneration.
> * Merge branch 'unstable' of github.com:rcpch/digital-growth-charts-server
> into unstable
> * some further tweaks to the Docker setup
> * readme additions for Docker in development
> * updates to build docker scfipt
> * create all charts and tims centiles
> * create charts
> * redo charts
> * create uk-who charts at runtime
> * autogenerated charts for all references on load
> * label to age_type in plottablechild
> * plottable child changes
> * update
> * comment out growth chart data generation on deploy
> * update measurement class return corrected calendar age
> * add measurement_method to plottable child return object
> * update readme
> * tidy corrected age
>
> File Changes
>
> * M .github/workflows/alpha_rcpch-dgc-server-alpha.yml<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-4c22d0de80c5087b669348076675fab6dbe0fa1a0ae0bd50cc18ea9f550e2cc8>
> (47)
> * M .github/workflows/unstable_rcpch-dgc-server-unstable.yml<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-4af47dd1cbee67644da9fbf6798e1f4b9ed6042787836ba4af57f2568fd9c12e>
> (47)
> * M .gitignore<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947>
> (5)
> * M Dockerfile<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557>
> (11)
> * M README.md<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
> (26)
> * A apispec_generation.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-382dcb73280c810dc9b42813fa8017ba933af740579114239926dd7b08443144>
> (94)
> * M app.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-568470d013cd12e4f388206520da39ab9a4e4c3c6b95846cbc281abc1ba3c959>
> (333)
> * M blueprints/__init__.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-6a871e951adf07b320d1c0babe4801a61437a6c3b7a0c9f6785293a616f104dd>
> (5)
> * A blueprints/openapi_blueprint.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-c0e40745bf2822d9ab689e60cbbbe89b882f0d273c15e24c3c81e430c71d8bee>
> (38)
> * A blueprints/uk_who_blueprint.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-ea1533b2dd06facc0987b6542876ecb95b69601019133f62806ee3c71df57f7b>
> (175)
> * M controllers/__init__.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-99acfc9e804ef000aafc8020f5f2c29802fc8b5c998fc4f2fc318a6fe2a0c1bf>
> (1)
> * M controllers/calculations.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-eeaf6080a25d7f14fc5932468d30ddcc6f30d847a271fcb58154c1af6c8b0ef9>
> (22)
> * D controllers/chart.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-d9e867723cf2f1ef5676c4430e107dba8b13d35887baaeefb0e389b2c4a207ba>
> (279)
> * M controllers/fictional_children.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-a38dded4b60b2b38f7fc82ebf116efa70a268fd0bb593dd47023e7778c274b8f>
> (6)
> * M controllers/plottable_child.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-94cd9cc55bc37ed7b77d8bb4f8916775cc2b3b6838b378a163fe7d73a7b427c6>
> (70)
> * M controllers/temp_test_functions.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-352ba0552a4bc711d7b9ec4bdc344e658edccc37073d77e1ddcd0b36b082168b>
> (8)
> * A controllers/uk_who_chart.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-9b499cecaf691f607347874de82b3c9f3ffdee70c834ebe8503189b5ab697de7>
> (32)
> * M controllers/upload.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-e14eeb08847460a7f55e934a3e21a8c17dd4e12b35bc3b55583620e4f17064b7>
> (240)
> * M docs/developer-documentation/README.md<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-0a843bb476fefe424de57bacbc0e6cfb5fd457b252ff3611eaaf8eb890953c55>
> (17)
> * A openapi.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-a9865368a7fc7fa33065e35b2343f10d08fb79d65205435403d0a163a3044713>
> (523)
> * R openapi.yml<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-5016bdc926b6c0f649f363739d09159a1db664aae19a08fa6b38b03406f040dc>
> (144)
> * M rcpchgrowth.code-workspace<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-41156f2e03b9f5ca0a1837449a790def4bae87eb38df30ed6160b9adf528b5d3>
> (7)
> * M rcpchgrowth/rcpchgrowth/__init__.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-f6bdded4fed72f92255b5bcf5c968a5f060660f17394527f2ec84e02bf25f164>
> (4)
> * M rcpchgrowth/rcpchgrowth/bmi_functions.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-5a260ef63e431bc535c5b4300358760c94d3d28c17464a470de608118dd3182a>
> (5)
> * M rcpchgrowth/rcpchgrowth/constants/__init__.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-e22bc786ac37e92a16dcfb83e85e3faa9152cd2e5f207aa95c508cf97a7be324>
> (1)
> * M rcpchgrowth/rcpchgrowth/constants/measurement_constants.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-9ed9951787b960d443706fc97526d94af0a5f7d035f78270217ed1fe2900a7b5>
> (74)
> * A rcpchgrowth/rcpchgrowth/constants/parameter_constants.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-e531335b8ed85f4c17ad11ccb9e405daf3116cd7debd571d638133787ab907bd>
> (22)
> * M rcpchgrowth/rcpchgrowth/constants/validation_constants.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-134b5894cf5b11d94c1e0045168f73827eaab205b604bcb3ccbba7bdb6ef18f4>
> (6)
> * M rcpchgrowth/rcpchgrowth/data_tables/data_tables.txt<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-0df77b164b86164412392f033654be93ae65c0fe3075414a787ae9605af853bd>
> (1)
> * M rcpchgrowth/rcpchgrowth/data_tables/trisomy_21.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-4596a2bef7c47cea4f40811f1d6f0ecfabef04ecb47d0426c37179c4cb0eb61a>
> (3803)
> * M rcpchgrowth/rcpchgrowth/data_tables/turner.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-e2e31993006eec798b3f187baa0d748a729ca499a0d8e51c1e7069e36a949d33>
> (43)
> * R rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/GrowthCharts.xls<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-84f4024f917b9dcbf3ac42b4df5afc0e6c0c25406097320b23410cffbed9068e>
> (0)
> * R rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/RCPCH weight
> correlation matrix by month.csv<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-212070e5c06f5b055df91065b02d945d4364ff8c5ee1c75ee33060df88756950>
> (0)
> * R rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/RCPCH weight
> correlation matrix by week.csv<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-9a5c14b3fffd9e35934911a50eee1d225fbdbe2c0d705c95bd8c0afe2cc03c4b>
> (0)
> * A
> rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/huiqi_cole_methods.bas<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-f07adfd45e7ade409988769d4ec0221830df953e87718c1737e2f9049a6b4b5b>
> (4933)
> * R
> rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/uk_who_0_20_preterm.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-fc6185a2d0a9e92380e7dffd3ae9b641a437d0732969dd8a1ca549f5291b31c5>
> (0)
> * R
> rcpchgrowth/rcpchgrowth/data_tables/uk-who_resources/uk_who_0_20_term.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-b60839751e0d9f292b81f3dd360a0d4d793463d73ef58bebffbfc87ffd64b07d>
> (0)
> * A rcpchgrowth/rcpchgrowth/data_tables/uk90_child.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-9c749f32d48bfea45ad9a7ae9fb358e60742a3a74227bc090f9f845dbb146672>
> (1517)
> * A rcpchgrowth/rcpchgrowth/data_tables/uk90_preterm.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-078a6a36e2981c9d37d5a6f25614cd09e1f60ad4e755de209466d9addd8ffc86>
> (189)
> * A rcpchgrowth/rcpchgrowth/data_tables/uk90_term.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-892ff9c5483cbed7e4fb22ca604fdafeaa8eed7bdbf9c737d48625aacc2863ae>
> (41)
> * D rcpchgrowth/rcpchgrowth/data_tables/uk_who.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-94ead9ef676dab12026cd6a9c24c5a440115359354ba139a318c636c1a84d5e4>
> (32)
> * D
> rcpchgrowth/rcpchgrowth/data_tables/uk_who_0_20_preterm_intervals_included.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-e1f0d1f2c54f48d60fc13df990aec541f75a21b6c158d808c06283593d1df14f>
> (2160)
> * D rcpchgrowth/rcpchgrowth/data_tables/uk_who_centiles.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-28e84629c89e6de5ddbde09bc1dbb3c745f83ef26aad2ca5ad292d2f55be160e>
> (0)
> * A rcpchgrowth/rcpchgrowth/data_tables/who_children.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-2723a52dd474ee88d932c508e4ec58f32b3245b0f78ba93febd5582a1550eadb>
> (233)
> * A rcpchgrowth/rcpchgrowth/data_tables/who_infants.json<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-5578d9a4d81a830bf38a4b48f7a61d26bea22d1820c10d58c585584391c5658a>
> (323)
> * M rcpchgrowth/rcpchgrowth/date_calculations.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-a4bdbd58a757a0fad02eaa661b491d7944859f1a999e8c744c5e0e434e219341>
> (28)
> * M rcpchgrowth/rcpchgrowth/dynamic_growth.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-a43be1b2fea5f8df4f3afe380f91a2ed3fad7b9e868cc71482ebc81911987fd2>
> (179)
> * M rcpchgrowth/rcpchgrowth/fictional_children.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-027d24fbe1ac44a24a7d62127d2ba684408642b0c20af166c61f33bb4466ebac>
> (5)
> * A rcpchgrowth/rcpchgrowth/global_functions.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-319f1da7c00841a162ee41ed3da11847d063cbaf7b912d386536ccddca057b8f>
> (648)
> * M rcpchgrowth/rcpchgrowth/measurement.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-1070ce6b9a9a5be3c3a02a7a664e1165657555392ebe15958ef8dda0ceb7fd29>
> (299)
> * A rcpchgrowth/rcpchgrowth/schemas/__init__.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-d971d62c38a7cbde6e9b11fb59e2e3738f4bdb98fec7909cdf9faf24d1341113>
> (1)
> * A rcpchgrowth/rcpchgrowth/schemas/measurement_class_schema.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-ec62ed2a261f99a7a13ab82549189ae9db98279ced92fbf7b42569ef5b136c47>
> (29)
> * D rcpchgrowth/rcpchgrowth/sds_calculations.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-320b5b8a7c7499189dece810c52f8af00e598be3eb978ddf7299713c4e0cb515>
> (483)
> * M rcpchgrowth/rcpchgrowth/tests/test_measurement_class.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-a09bbf79fc7515be0365e7892936b39db734c25e47df5ab6d042c759d645a986>
> (32)
> * A rcpchgrowth/rcpchgrowth/trisomy_21.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-8b78a41727c2adaa37b79639ff3177148d5dcbdcf9fe05d7092fe65000493dcc>
> (75)
> * A rcpchgrowth/rcpchgrowth/turner.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-f654fb5d6ea53476a05d1088eeeb556c8feb7e6223aca922b967947b1f3ee15a>
> (77)
> * A rcpchgrowth/rcpchgrowth/uk_who.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-d432f60104a0f126a8158e08c21fcff35d3f1c0a9444a1d2385fe28608c56b11>
> (216)
> * M rcpchgrowth/setup.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-245bf863fb42663de9544caa6bd82cb585ff0cf03074f0bf8c78454a4fc3375d>
> (7)
> * M requirements.txt<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552>
> (6)
> * A s/rebuild-docker<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-d0bcee849b6beac8ad46d6866e54ccff6b87f091cafdbebcaa84f00717a00193>
> (20)
> * M s/start-docker<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-5e3fbf896a54e69f44a77ddcff819be82ef17c0be7ff4292b98035e2b4123ff8>
> (14)
> * M schemas/__init__.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-1d41be12fbded48f0ac94e7231190b1752403326c3d0bf98792522a1162be0f0>
> (2)
> * R schemas/calculation_schemas.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-42091a942d3be09918b5880dae4122d069fbe9656ae258a080b30d72130d51a5>
> (13)
> * M schemas/measurement_schemas.py<
> https://github.com/rcpch/digital-growth-charts-server/pull/129/files#diff-ea2ef5c4f0cdcb5ecbe8dc31960e522324084b5abebf10cc891311f1d83af462>
> (2)
>
> Patch Links:
>
> * https://github.com/rcpch/digital-growth-charts-server/pull/129.patch
> * https://github.com/rcpch/digital-growth-charts-server/pull/129.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub<
> https://github.com/rcpch/digital-growth-charts-server/pull/129>, or
> unsubscribe<
> https://github.com/notifications/unsubscribe-auth/ABGSJJIM54SNTSTXWQWSZADS5U4IDANCNFSM4XGII6EA>.
>
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/rcpch/digital-growth-charts-server/pull/129#issuecomment-774476830>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAR5KYLBNXF65YUO2YGQX2LS5U7ALANCNFSM4XGII6EA>
> .
>
